release-team/README.md

```diff
@@ -188,6 +188,60 @@ For all unplanned or embargoed releases
 
 ---
 
+## Release Team Selection
```

Comment:
nit: can we get a ToC at the top of this document? :)

Reply:
@nikhita -- agreed that it's a long doc! I've added a short ToC for the high-level things and there's also https://github.com/kubernetes/sig-release/issues/238 opened for cleaning up the role definitions. :)